### PR TITLE
Feature/s3 c 9226 create bucket

### PR DESCRIPTION
**Pull request by anurag4DSB**
feature: Create Bucket using provisioner
- Updated provisioner to implement create bucket functionality
- Added an S3 client to interact with S3
- Added logs to the methods not implemented yet
- Added placeholder for accepting TLS certificates for S3 endpoint.
- Added mock clients and tests for provisioner

### CODE REVIEW

File: pkg/util/s3client/s3-handler.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
+/*
```
Comment (fredmnl) [inline pkg/util/s3client/s3-handler.go:1]: Should this be in a `LICENSE` file?

File: pkg/util/s3client/mock/s3iface_mock.go [hunk 2]
```diff
@@ -0,0 +1,5317 @@
+// Code generated by MockGen. DO NOT EDIT.
```
Comment (fredmnl) [inline pkg/util/s3client/mock/s3iface_mock.go:1]: We should at least document how this is generated, for repeatability. How about adding it in the build step of the Makefile?

File: pkg/driver/provisioner.go [hunk 3]
```diff
@@ -112,7 +159,74 @@ func (s *provisionerServer) DriverGrantBucketAccess(ctx context.Context,
...
+	s3Client, err := s3client.InitS3Client(accessKey, secretKey, endpoint, region, nil, true)
+	if err != nil {
+		klog.ErrorS(err, "Failed to create S3 client", "endpoint", endpoint)
+		return nil, status.Error(codes.Internal, "failed to create S3 client")
+	}
+	klog.V(3).InfoS("Successfully initialized S3 client", "endpoint", endpoint)
+	return s3Client, nil
+}
+
+func fetchS3Parameters(secretData map[string][]byte) (string, string, string, string, string, error) {
```
Comment (jonathan-gramain) [inline pkg/driver/provisioner.go:202]: May be worth creating a private `type s3Params struct{ AccessKey string ...}` returned by this function.

File: pkg/util/s3client/s3-handler.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+	if creationErr != nil {
+		return creationErr
+	}
+
+	klog.InfoS("Bucket creation succeeded", "bucketName", bucketName)
+	return nil
+}
+
+// configureTLSTransport sets up the HTTP transport with TLS support
+func configureTLSTransport(certData []byte, skipTLS bool) *http.Transport {
```
Comment (jonathan-gramain) [inline pkg/util/s3client/s3-handler.go:108]:
```suggestion
func configureTLSTransport(certData []byte, skipVerify bool) *http.Transport {
```

**Review by jonathan-gramain — APPROVED** [review]

**Comment by anurag4DSB** [comment]
As too much rework is needed, I am gonna divide this into 2 PRs
Major rework needed:
- Use AWS SDK v2
- Use Ginkgo for tests


First PR: https://github.com/scality/cosi/pull/6

**Pull request closed without merging**